src/Native/Unix/System.Native/pal_networking.c

```diff
@@ -142,11 +142,19 @@ c_static_assert(GetHostErrorCodes_NO_DATA == NO_DATA);
 c_static_assert(GetHostErrorCodes_NO_ADDRESS == NO_ADDRESS);
 c_static_assert(sizeof(uint8_t) == sizeof(char)); // We make casts from uint8_t to char so make sure it's legal
 
+// sizeof_member(struct foo, bar) is not valid C++.
+// The fix is to remove struct. That is not valid C.
+// Use typedefs to make it valid C -- which are redundant but valid C++.
+typedef struct iovec iovec;
+typedef struct sockaddr sockaddr;
+typedef struct xsocket xsocket;
+typedef struct linger linger;
+
 // We require that IOVector have the same layout as iovec.
 c_static_assert(sizeof(IOVector) == sizeof(struct iovec));
-c_static_assert(sizeof_member(IOVector, Base) == sizeof_member(struct iovec, iov_base));
+c_static_assert(sizeof_member(IOVector, Base) == sizeof_member(iovec, iov_base));
 c_static_assert(offsetof(IOVector, Base) == offsetof(struct iovec, iov_base));
```

Comment:
It would be nice to drop `struct` for offsetof here as well to make this look symmetric.

Reply:
Agreed, I just hadn’t checked for sufficiently typedefs, which appear to be there, or if not would be trivial. Will do.